ballerina/main.bal

```diff
@@ -83,22 +83,26 @@ isolated function buildPromptString(Prompt prompt) returns string {
     return str.trim();
 }
 
-isolated function getPromptWithExpectedResponseSchema(string prompt, map<json> expectedResponseSchema) returns string =>
-    string `${prompt}
+isolated function getPromptWithExpectedResponseSchema(Prompt prompt, typedesc<anydata> expectedResponseTypedesc) returns string =>
+    string `${buildPromptString(prompt)}
         ---
 
         The output should be a JSON value that satisfies the following JSON schema, 
         returned within a markdown snippet enclosed within ${"```json"} and ${"```"}
         
         Schema:
-        ${expectedResponseSchema.toJsonString()}`;
+        ${getExpectedResponseSchema(expectedResponseTypedesc).toJsonString()}`;
 
-isolated function callLlmGeneric(Prompt prompt, Context context, typedesc<json> targetType,
-                                 map<json>? jsonSchema) returns json|error {
+isolated function callLlmGeneric(Prompt prompt, Context context, 
+                                 typedesc<anydata> expectedResponseTypedesc) returns anydata|error {
     Model model = context.model;
-    json resp =
-         check model->call(buildPromptString(prompt), jsonSchema ?: generateJsonSchemaForTypedescAsJson(targetType));
-    return parseResponseAsType(resp, targetType);
+    anydata response = check model->call(prompt, expectedResponseTypedesc);
+    anydata|error result = response.ensureType(expectedResponseTypedesc);
+    if result is error {
+        return error(string `Invalid value returned from the LLM Client, expected: ${
```

Comment:
Shall we add the original error message too.
It will help the user to identify the issue

Reply:
This is an ensureType error. What additional information would it contain that is not covered in this error?